src/main/java/uk/gov/pay/directdebit/webhook/gocardless/services/WebhookGoCardlessService.java

```diff
@@ -28,10 +29,15 @@
 
 import static java.lang.String.format;
 import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.partitioningBy;
+import static uk.gov.pay.directdebit.events.model.GoCardlessResourceType.MANDATES;
+import static uk.gov.pay.directdebit.events.model.GoCardlessResourceType.PAYMENTS;
+import static uk.gov.pay.directdebit.events.model.GoCardlessResourceType.PAYOUTS;
 
 public class WebhookGoCardlessService {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(WebhookGoCardlessService.class);
+    private static final List<GoCardlessResourceType> VALID_RESOURCE_TYPES = List.of(PAYMENTS, MANDATES, PAYOUTS);
```

Comment:
This could be a `Set` (or even an `EnumSet`!)

Reply:
It probably doesn’t make much difference in this case but generally (particularly with large collections), a `Set` will have faster lookups because it will use hashing (or bit field magic if it’s an `EnumSet`) rather than linearly looking through every element in the list.

Reply:
Makes sense. I've deleted this line though, as it's no longer needed anyway!